tensorflow/core/platform/env.h

```diff
@@ -218,6 +218,10 @@ class Env {
   /// Creates a local unique temporary file name. Returns true if success.
   bool LocalTempFilename(string* filename);
 
+  /// Create a unique file name starts with |prefix| and ends with |suffix|.
+  /// Returns true if success.
+  bool CreateTempFilename(string* prefix, const string& suffix);
```

Comment:
How is this different from CreateLocalFilename?

Maybe what you actually want to do is add an optional suffix param to CreateLocalFilename?  Or even just call CreateLocalFilename inside of hlo_graph_dumper and just append the suffix to the returned filename?

Reply:
> How is this different from CreateLocalFilename? (I mean, I can tell from reading the code, but not really from reading the header.)

Changed name to `CreateLocalFilename`.

> Maybe what you actually want to do is add an optional suffix param to CreateLocalFilename?

You mean `bool CreateTempFilename(string* prefix, const string& suffix = "")`?

> Or even just call CreateLocalFilename inside of hlo_graph_dumper and just append the suffix to the returned filename?

Then we will have to check if the new filename actually exists after appending suffix, repeating the work of `CreateLocalFilename`.

Reply:
Sorry, I think I reviewed this before I saw this comment.

Anyway I like what you came up with.  :)